### PR TITLE
Refactor ban command

### PR DESCRIPTION
**Pull request by DevSpen**

### CODE REVIEW

File: src/commands/moderation/ban.ts [hunk 1]
```diff
@@ -8,27 +8,24 @@ import type { CommandInteraction, Guild, GuildMember, User } from 'discord.js';
...
-		this.guild = guild;
+		const guild = interaction.guild!;
 
 		const requestedUser = interaction.options.getString('user', true);
 		const reason = interaction.options.getString('reason', false) ?? undefined;
 		const days = interaction.options.getInteger('days_to_delete', false) ?? 0;
 		if (requestedUser.includes(',')) {
 			interaction.reply('Banning users...');
-			const requestedUsers = [...new Set(requestedUser.split(/\s*,\s*/g).map((el) => this.parseUser(el)))];
+			const requestedUsers = [...new Set(requestedUser.split(/\s*,\s*/g).map((el) => this.parseUser(el, guild)))];
```
Comment (greysilly7) [inline src/commands/moderation/ban.ts:23]: Please be more descriptive on the naming of "el"

File: src/commands/moderation/ban.ts [hunk 2]
```diff
@@ -8,27 +8,24 @@ import type { CommandInteraction, Guild, GuildMember, User } from 'discord.js';
...
 			let invalidUsers: string[] = [];
 			let failedOperations: [user: ResolvedUser, error: any][] = [];
 
 			for (let i = 0; i < requestedUsers.length; i++) {
```
Comment (greysilly7) [inline src/commands/moderation/ban.ts:27]: please use `for (const x of y) {}`

File: src/commands/moderation/ban.ts [hunk 3]
```diff
@@ -99,7 +96,7 @@ export class BanCommand extends Command {
 			});
 	}
 
-	private async parseUser(el: string): Promise<ResolvedUser> {
+	private async parseUser(el: string, guild: Guild): Promise<ResolvedUser> {
```
Comment (greysilly7) [inline src/commands/moderation/ban.ts:99]: Again more descriptive of what "el" is

**Comment by DevSpen** [comment]
Decided to remove the ability to ban multiple users, will probably make a separate command for that. It got really messy trying to combine it.

File: src/commands/moderation/ban.ts [hunk 4]
```diff
@@ -96,38 +41,19 @@ export class BanCommand extends Command {
...
 			builder
 				.setName(this.name)
 				.setDescription(this.description)
-				.addStringOption((builder) =>
-					builder
-						.setName('user')
-						.setRequired(true)
-						.setDescription("The user(s) to ban. If you'd like to ban multiple users, separate them using commas.")
-				)
+				.addUserOption((builder) => builder.setName('user').setRequired(true).setDescription('The user to ban.'))
```
Comment (greysilly7) [inline src/commands/moderation/ban.ts:55]: change from "user" to "userToBan"

File: src/commands/moderation/ban.ts [hunk 5]
```diff
@@ -1,91 +1,36 @@
...
+		if (!interaction.memberPermissions!.has(Permissions.FLAGS.BAN_MEMBERS))
+			new UserError(interaction)
+				.setResponse({ embeds: [generateErrorEmbed('You need the ban members permission to use that.', 'Missing Permissions')] })
+				.setType('MISSING_PERMISSIONS')
+				.sendResponse();
+
+		const guild = interaction.guild!;
 
-		const requestedUser = interaction.options.getString('user', true);
+		const user = interaction.options.getUser('user', true);
```
Comment (greysilly7) [inline src/commands/moderation/ban.ts:23]: change from "user" to "userToBan"

**Review by greysilly7 — CHANGES_REQUESTED** [review]

**Review by greysilly7 — APPROVED** [review]

**Pull request merged**